scripts/check-component-docs.sh

```diff
@@ -0,0 +1,22 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+# check-component-docs.sh
+#
+# SUMMARY
+#
+#   Checks that there are no changed machine-generated component Cue files after running the
+#   generatation step via `make generate-component-docs`.
+#
+#   This should only be run in CI, after calling `make generate-component-docs`, as it depends on
+#   the presence of dirty files in the working directory to detect if the machine-generated files
+#   are out-of-date.
```

Comment:
:thinking: 
Just wondering if there is a way for this script to know if it is executed outside of CI, and exit with a message?

Reply:
There's no reason to run this script itself outside of CI, since you achieve the same effect by just running `make generate-component-docs` which is what it would already do if you tried to run `make check-component-docs`.

Reply:
But also maybe that is overkill if nothing "bad" would happen by it executing outside of CI.

Reply:
Yeah, I mean, the worst that happens is a false positive because you did update the files but haven't committed them yet.